src/json.rs

```diff
@@ -65,13 +65,13 @@ impl JsonValue {
         }
     }
 
-    pub fn u128_from_str(&self, key: &str) -> Result<u128, ()> {
+    pub fn u128_from_str(&self, key: &str) -> Result<u128, JsonError> {
         match self {
-            JsonValue::Object(o) => match o.get(key).ok_or(())? {
-                JsonValue::String(s) => s.parse::<u128>().map_err(|_| ()),
-                _ => Err(()),
+            JsonValue::Object(o) => match o.get(key).ok_or(JsonError::MissingValue)? {
+                JsonValue::String(s) => s.parse::<u128>().map_err(|_| JsonError::InvalidU128),
+                _ => Err(JsonError::InvalidU128),
             },
-            _ => Err(()),
+            _ => Err(JsonError::NotJsonType),
         }
```

Comment:
This function should be removed. `u128` already handles this thing indeed.

Reply:
removed